python/ql/src/Security/CWE-022/PathInjection.ql

```diff
@@ -25,7 +26,8 @@ import semmle.python.web.HttpRequest
 import semmle.python.security.injection.Path
 
 
-from TaintSource src, TaintSink sink
-where src.flowsToSink(sink)
 
-select sink, "This path depends on $@.", src, "a user-provided value"
+from TaintedNode srcnode, TaintedNode sinknode, TaintSource src, TaintSink sink
+where src.flowsToSink(sink) and srcnode.getNode() = src and sinknode.getNode() = sink
```

Comment:
I feel like the `srcnode.getNode() = src and sinknode.getNode() = sink` construction is a bit clunky. Could we make this into a method (`.getSource()`) on `TaintedNodes`, etc. and then simply use `srcnode.getSource()` in the `select` statement?

Reply:
That makes sense. I've changed the general form from
```
from TaintedNode srcnode, TaintedNode sinknode, TaintSource src, TaintSink sink
where src.flowsToSink(sink) and srcnode.getNode() = src and sinknode.getNode() = sink
select sink, srcnode, sinknode, ..., src, ...
```
to 
```
from TaintedPathSource src, TaintedPathSink sink
where src.flowsTo(sink)
select sink.getSink(), src, sink,..., src.getSource(), ...
```

Reply:
Perfect!